src/vdf_io/export_vdf/pinecone_export.py

```diff
@@ -232,172 +231,11 @@ def get_all_ids_from_index(self, num_dimensions, namespace="", hash_value=""):
-                )
-                prev_size = len(all_ids)
-                all_ids.update(ids)
-                if len(ids) == 0:
-                    tqdm.write("No new ids found, exiting...")
-                    break
-                new_ids = all_ids - set(ids)
-                if len(new_ids) > 0:
-                    # fetch 1 random vector from new_ids
-                    self.update_range_from_new_ids(
-                        vector_range_min, vector_range_max, new_ids, namespace
-                    )
-                curr_size = len(all_ids)
-                if curr_size > prev_size:
-                    tqdm.write(
-                        f"updating ids set with {curr_size - prev_size} new ids..."
-                    )
-                try_count += 1
-                if try_count > max_tries and len(all_ids) < num_vectors:
-                    tqdm.write(
-                        f"Could not collect all ids after {max_tries} random searches."
-                        " Please provide range of ids instead. Exporting the ids collected so far."
-                    )
-                    break
-                pbar.update(curr_size - prev_size)
-        tqdm.write(
-            f"Collected {len(all_ids)} ids out of {num_vectors} vectors in {try_count} tries."
-        )
-        return all_ids
-
-    def update_range_from_new_ids(
-        self, vector_range_min, vector_range_max, new_ids, namespace
-    ):
-        # use update_range to update the range of the vectors
-        self.update_range(
-            new_ids, vector_range_min, vector_range_max, namespace, size=1
-        )
-
-    def update_range(
-        self, all_ids, vector_range_min, vector_range_max, namespace, size=10
-    ):
-        random_ids = np.random.choice(list(all_ids), size=size).tolist()
-
-        random_vectors = [
-            x["values"]
-            for x in self.index.fetch(random_ids, namespace=namespace)[
-                "vectors"
-            ].values()
-        ]
-        # extend the range of the vectors
-        random_vectors_np = np.array(random_vectors)
 
-        # Initialize vector_range_min and vector_range_max if they are not set
-        if vector_range_min is None:
-            vector_range_min = np.min(random_vectors_np, axis=0)
-        else:
-            vector_range_min = np.minimum(
-                vector_range_min, np.min(random_vectors_np, axis=0)
-            )
-
-        if vector_range_max is None:
-            vector_range_max = np.max(random_vectors_np, axis=0)
-        else:
-            vector_range_max = np.maximum(
-                vector_range_max, np.max(random_vectors_np, axis=0)
-            )
-
-        return vector_range_min, vector_range_max
+        # Use list_points to get all IDs directly
+        all_ids = list(self.index.list_points(namespace=namespace))
+        tqdm.write(f"Collected {len(all_ids)} IDs using list_points.")
+        return all_ids
```

Comment:
This is the documentation for the list_point functionality:

Paginate through results
The list operation returns up to 100 IDs per page at a time by default. If the limit parameter is passed, list returns up to that number of IDs per page instead. For example, if limit=3, up to 3 IDs be returned per page. Whenever there are additional IDs to return, the response also includes a pagination_token for fetching the next page of IDs.

[​](https://docs.pinecone.io/guides/data/list-record-ids#implicit-pagination)
Implicit pagination
When using the Python client, list paginates automatically.

Python

from pinecone import Pinecone

pc = Pinecone(api_key='YOUR_API_KEY')
index = pc.Index("pinecone-index")

for ids in index.list(namespace='ns1'):
    print(ids)

# Response:
# ['doc1#chunk1', 'doc1#chunk2', 'doc1#chunk3']
# ['doc1#chunk4', 'doc1#chunk5', 'doc1#chunk6']
# ...
[​](https://docs.pinecone.io/guides/data/list-record-ids#manual-pagination)
Manual pagination
To manually fetch each page of results yourself, first use the list_paginated() (Python), listPaginated (Node.js), or the REST API directly:


from pinecone import Pinecone

pc = Pinecone(api_key='YOUR_API_KEY')
index = pc.Index("pinecone-index")

namespace = 'ns1'

# For manual control over pagination
results = index.list_paginated(
    prefix='pref',
    limit=3,
    namespace='ns1'
)
print(results.namespace)
print([v.id for v in results.vectors])
print(results.pagination.next)
print(results.usage)

# Results:
# ['10103-0', '10103-1', '10103-10']
# eyJza2lwX3Bhc3QiOiIxMDEwMy0=
# {'read_units': 1}

Then, to get the next batch of IDs, use the returned pagination_token:


from pinecone import Pinecone

pc = Pinecone(api_key='YOUR_API_KEY')
index = pc.Index("pinecone-index")

namespace = 'ns1'

results = index.list_paginated(
    prefix='pref',
    limit=3,
    namespace='ns1',
    pagination_token='eyJza2lwX3Bhc3QiOiIxMDEwMy0='
)
print(results.namespace)
print([v.id for v in results.vectors])
print(results.pagination.next)
print(results.usage)

# Response:
# ['10103-0', '10103-1', '10103-10']
# xndlsInByZWZpeCI6IjEwMTAzIn0==
# {'read_units': 1}
When there are no more IDs to return, the response does not includes a pagination_token:



from pinecone import Pinecone

pc = Pinecone(api_key='YOUR_API_KEY')
index = pc.Index("pinecone-index")

namespace = 'ns1'

results = index.list_paginated(
    prefix='10103',
    limit=3,
    pagination_token='xndlsInByZWZpeCI6IjEwMTAzIn0=='
)

print(results.namespace)
print([v.id for v in results.vectors])
print(results.pagination.next)
print(results.usage)

# Response:
# ['10103-4', '10103-5', '10103-6']
# {'read_units': 1}

Read this to properly use the functionality for getting the ids